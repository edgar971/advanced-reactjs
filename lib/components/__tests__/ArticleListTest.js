import React from 'react';
import renderer from 'react-test-renderer';
import ArticleList from '../ArticleList';

describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: { id: 'a', title: 'best title a'},
      b: { id: 'b', title: 'best title b'}
    }, 
    articleActions: {
      lookupAuthor: jest.fn(() => ({}))
    }
  };

  it('renders correctly', () => {
    const tree = renderer.create(<ArticleList {...testProps} />).toJSON();
    
    expect(tree.children.length).toBe(2);
    expect(tree).toMatchSnapshot();
  });
});