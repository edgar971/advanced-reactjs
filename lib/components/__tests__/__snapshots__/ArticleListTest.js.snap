// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "gray",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <h3
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      best title a
    </h3>
    <p
      style={
        Object {
          "fontSize": ".85em",
        }
      }
    >
      <small>
        Invalid Date
      </small>
    </p>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      >
        undefined undefined
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      <p />
    </div>
  </div>
  <div
    style={
      Object {
        "borderBottomColor": "gray",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    <h3
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      best title b
    </h3>
    <p
      style={
        Object {
          "fontSize": ".85em",
        }
      }
    >
      <small>
        Invalid Date
      </small>
    </p>
    <div
      style={
        Object {
          "paddingBottom": 10,
          "paddingTop": 10,
        }
      }
    >
      <a
        href={undefined}
      >
        undefined undefined
      </a>
    </div>
    <div
      style={
        Object {
          "paddingLeft": 20,
        }
      }
    >
      <p />
    </div>
  </div>
</div>
`;
